### PR TITLE
Use human-readable stat var names in tile legends

### PR DESCRIPTION
**Pull request by juliawu**
Updates the behavior of all our tiles with legends, to fetch the stat var name with an api call if it's not provided in a statVarSpec. If no name is found in the KG, will default to using the stat's dcid.

Useful as a fallback if we don't specify a name in configs, and makes our web component legend labels more user-friendly.

## Before:
Screenshot 2023-07-24 at 3 30 48 PM
Screenshot 2023-07-24 at 3 39 42 PM
Screenshot 2023-07-24 at 3 42 05 PM
Screenshot 2023-07-24 at 3 32 14 PM

## After:
Screenshot 2023-07-24 at 3 29 46 PM
Screenshot 2023-07-24 at 3 29 34 PM

Screenshot 2023-07-24 at 3 30 00 PM

### CODE REVIEW

**Review by chejennifer — COMMENTED** [review]
nice!

File: static/js/components/tiles/line_tile.tsx [hunk 1]
```diff
@@ -228,19 +230,19 @@ function rawToChart(
         const dataPoints: DataPoint[] = [];
         for (const obs of obsList) {
           dataPoints.push({
-            label: obs.date,
+            label: statVarDcidToName[spec.statVar],
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/line_tile.tsx]: Looks like this particular label doesn't get used, but reverted this change.

File: static/js/components/tiles/donut_tile.tsx [hunk 2]
```diff
@@ -191,7 +202,7 @@ function rawToChart(
       }
       const stat = raw.data[statVar][placeDcid];
       const dataPoint = {
-        label: getStatVarName(statVar, props.statVarSpec),
+        label: statVarNames[statVar],
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/components/tiles/donut_tile.tsx:205]: Good idea. Added a call to getStatsVarLabel to preserve that behavior.

File: static/js/utils/tile_utils.tsx [hunk 3]
```diff
@@ -75,6 +77,58 @@ export function getStatVarName(
...
+  apiRoot?: string
+): Promise<{ [key: string]: string }> {
+  if (_.isEmpty(statVarSpec)) {
+    return Promise.resolve({});
+  }
+  const statVarDcids: string[] = [];
+  const statVarNames: Record<string, string> = {};
+  // If a name is already provided by statVarSpec, use that name
+  statVarSpec.forEach((spec) => {
+    if (spec.name && spec.name !== "") {
```
[the remark this answers is not in the document]
Comment (juliawu) [reply static/js/utils/tile_utils.tsx]: Thanks for the pointer! Changed.

**Review by chejennifer — APPROVED** [review]
thanks for updating!

**Pull request merged**